crates/bitwarden-crypto/src/enc_string/symmetric.rs

```diff
@@ -159,16 +180,27 @@ impl EncString {
 
 impl Display for EncString {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        let parts: Vec<&[u8]> = match self {
-            EncString::AesCbc256_B64 { iv, data } => vec![iv, data],
-            EncString::AesCbc256_HmacSha256_B64 { iv, mac, data } => vec![iv, data, mac],
-        };
+        match self {
+            EncString::AesCbc256_B64 { .. } | EncString::AesCbc256_HmacSha256_B64 { .. } => {
+                let parts: Vec<&[u8]> = match self {
+                    EncString::AesCbc256_B64 { iv, data } => vec![iv, data],
+                    EncString::AesCbc256_HmacSha256_B64 { iv, mac, data } => vec![iv, data, mac],
+                    _ => unreachable!(),
+                };
 
-        let encoded_parts: Vec<String> = parts.iter().map(|part| STANDARD.encode(part)).collect();
+                let encoded_parts: Vec<String> =
+                    parts.iter().map(|part| STANDARD.encode(part)).collect();
 
-        write!(f, "{}.{}", self.enc_type(), encoded_parts.join("|"))?;
+                write!(f, "{}.{}", self.enc_type(), encoded_parts.join("|"))?;
 
-        Ok(())
+                Ok(())
+            }
+            EncString::XChaCha20_Poly1305_Cose_B64 { data } => {
+                write!(f, "{}.{}", self.enc_type(), STANDARD.encode(data))?;
```

Comment:
Is there not a more human-readable way to present this? I would expect cbor to have some kind of to_json method that would represent key-value pairs nicely and (hopefully) b64 binary value-data. Maybe ciborium already has a display impl?

Reply:
much better, thank you!